ruma-events/src/lib.rs

```diff
@@ -234,17 +237,24 @@ pub trait MessageEventContent: RoomEventContent {}
 /// Marker trait for the content of a state event.
 pub trait StateEventContent: RoomEventContent {}
 
-/// Helper struct to obtain the event type from a serde_json::value::RawValue.
+/// Helper struct to determine the event kind from a serde_json::value::RawValue.
 #[doc(hidden)]
 #[derive(Debug, Deserialize)]
 pub struct EventDeHelper {
+    /// the Matrix event type string "m.room.whatever".
     #[serde(rename = "type")]
     pub ev_type: String,
-    #[serde(default)]
-    pub state_key: Option<String>,
-    #[serde(default)]
-    pub event_id: Option<String>,
+
+    /// If present the event is a state event.
+    pub state_key: Option<IgnoredAny>,
+
+    /// If no `state_key` is found but an `event_id` is present the a message event is deserialized.
```

Comment:
Please re-read and make the comments of `state_key`, `event_id` and `room_id` consistent which each other.